client/me/notification-settings/blogs-settings/header.jsx

```diff
@@ -45,29 +44,31 @@ class BlogSettingsHeader extends PureComponent {
 	};
 
 	getLegend = () => {
-		const tally = o => o.reduce( ( total, value ) => total + value );
-		const sizeAndSum = settings => [ settings.size, tally( settings ) ];
-
-		let counts = this.props.settings
-			.deleteIn( [ 'email', 'achievement' ] )
-			.filterNot( ( _, key ) => includes( [ 'blog_id', 'devices' ], key ) )
-			.map( sizeAndSum )
-			.toArray();
-
-		counts = this.props.settings
-			.get( 'devices' )
-			.map( device => device.filter( ( _, key ) => key !== 'device_id' ) )
-			.map( sizeAndSum )
-			.toArray()
-			.concat( counts );
-
-		const [ size, count ] = zip.apply( null, counts ).map( tally );
+		const { settings } = this.props;
+
+		// Ignore blog_id, email.achievement and devices (we'll handle devices separately).
+		const filteredSettings = {
+			...omit( settings, [ 'blog_id', 'devices' ] ),
+			email: omit( get( settings, 'email', {} ), 'achievement' ),
+		};
+		// Ignore the device_id of each device found.
+		const devicesSettings = map( settings.devices, device => omit( device, 'device_id' ) );
+
+		const [ onSettings, offSettings ] = partition(
+			flatten( [ ...map( filteredSettings, values ), ...map( devicesSettings, values ) ] )
+		);
 
-		if ( count === 0 ) {
+		if ( isEmpty( onSettings ) ) {
+			// TODO: currently it's not possible to reach 0, even with all checkboxes unchecked.
+			// 		 the two settings that don't have corresponding checkboxes are:
+			// 		 - timeline.store_order
+			// 		 - email.store_order
+			// 		 - email.scheduled_publicize
+			//		 Should we filter these also?
```

Comment:
Notifications exceptions are defined [here](https://github.com/Automattic/wp-calypso/blob/b7f55a75c65276e90260fdb8e3421be62ae5f230/client/me/notification-settings/settings-form/constants.js#L5). Should we use this constant also to filter notification types in the `getLegend` function?

`timeline.store_order` is a valid notification, but only present on sites that have a Woo store. See #19196 -- a small PR where the `store_order` notification is added.

I don't know why some notification emails are disabled, namely Achievements, Publicize events and Store Orders. Pinging people who worked on the more recent additions: @artpi (publicize) and @justinshreve (store).

Reply:
Store order emails are disabled because WooCommerce currently handles and sends out its own suite of email notifications. These are not tied in with WordPress.com's email/notification system, and we didn't want to implement double notifications.

Reply:
Thanks for the info @justinshreve - for now I'll update the logic to omit those items. The only consequence in this section is the copy shown in the section header. I'll leave a ref to this comment thread too.